tsparser/src/parser/types/typ.rs

```diff
@@ -137,6 +168,27 @@ pub enum Basic {
     Never,
 }
 
+impl Display for Basic {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s: &'static str = match self {
+            Basic::Any => "any",
+            Basic::String => "string",
+            Basic::Boolean => "boolean",
+            Basic::Number => "number",
+            Basic::Object => "object",
+            Basic::BigInt => "bigint",
+            Basic::Date => "Date",
```

Comment:
```suggestion
            Basic::Date => "date",
```

all the others are lowercase, but maybe this one is special?

Reply:
its special!